datafusion/expr/src/signature.rs

```diff
@@ -116,6 +116,12 @@ pub enum TypeSignature {
     /// Function `make_array` takes 0 or more arguments with arbitrary types, its `TypeSignature`
     /// is `OneOf(vec![Any(0), VariadicAny])`.
     OneOf(Vec<TypeSignature>),
+    /// Specifies Signatures for array functions
+    ArraySignature(ArrayFunctionSignature),
```

Comment:
For future extension

Reply:
I think that is a good change, though a change of the API